src/dependencies/dss-direct-deposit/D3MInit.sol

```diff
@@ -0,0 +1,326 @@
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_POOL"))), d3m.pool);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_PLAN"))), d3m.plan);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_ORACLE"))), d3m.oracle);
+    }
+
+    /**
+     * @dev This works for V2 and V3 pool adapters.
+     */
+    function initAavePool(
+        DssInstance memory dss,
+        D3MInstance memory d3m,
+        D3MCommonConfig memory cfg,
+        D3MAavePoolConfig memory aaveCfg
+    ) internal {
+        D3MAavePoolLike pool = D3MAavePoolLike(d3m.pool);
+
+        // Sanity checks
+        require(pool.hub() == cfg.hub, "Pool hub mismatch");
+        require(pool.ilk() == cfg.ilk, "Pool ilk mismatch");
+        require(pool.vat() == address(dss.vat), "Pool vat mismatch");
+        require(pool.dai() == address(dss.dai), "Pool dai mismatch");
+        require(pool.adai() == aaveCfg.adai, "Pool adai mismatch");
+        require(pool.stableDebt() == aaveCfg.stableDebt, "Pool stableDebt mismatch");
+        require(pool.variableDebt() == aaveCfg.variableDebt, "Pool variableDebt mismatch");
+
+        pool.file("king", aaveCfg.king);
+    }
+
+    function initCompoundPool(
+        DssInstance memory dss,
+        D3MInstance memory d3m,
+        D3MCommonConfig memory cfg,
+        D3MCompoundPoolConfig memory compoundCfg
+    ) internal {
+        D3MCompoundPoolLike pool = D3MCompoundPoolLike(d3m.pool);
+
+        // Sanity checks
+        require(pool.hub() == cfg.hub, "Pool hub mismatch");
+        require(pool.ilk() == cfg.ilk, "Pool ilk mismatch");
+        require(pool.vat() == address(dss.vat), "Pool vat mismatch");
+        require(pool.dai() == address(dss.dai), "Pool dai mismatch");
+        require(pool.comptroller() == compoundCfg.comptroller, "Pool comptroller mismatch");
+        require(pool.comp() == compoundCfg.comp, "Pool comp mismatch");
+        require(pool.cDai() == compoundCfg.cdai, "Pool cDai mismatch");
+
+        pool.file("king", compoundCfg.king);
+    }
+
+    function initAaveRateTargetPlan(
+        D3MInstance memory d3m,
+        D3MAaveRateTargetPlanConfig memory aaveCfg
+    ) internal {
+        D3MAaveRateTargetPlanLike plan = D3MAaveRateTargetPlanLike(d3m.plan);
+        ADaiLike adai = ADaiLike(aaveCfg.adai);
+
+        // Sanity checks
+        require(plan.adai() == address(adai), "Plan adai mismatch");
+        require(plan.stableDebt() == aaveCfg.stableDebt, "Plan stableDebt mismatch");
+        require(plan.variableDebt() == aaveCfg.variableDebt, "Plan variableDebt mismatch");
+        require(plan.tack() == aaveCfg.tack, "Plan tack mismatch");
+        require(plan.adaiRevision() == aaveCfg.adaiRevision, "Plan adaiRevision mismatch");
+        require(adai.ATOKEN_REVISION() == aaveCfg.adaiRevision, "ADai adaiRevision mismatch");
+
+        plan.file("bar", aaveCfg.bar);
+    }
```

Comment:
This function apparently is not used. Could potentially be removed.

Reply:
Same as above.